docusaurus/docs/iOS/03-guides/06-querying-calls.mdx

```diff
@@ -77,19 +77,71 @@ You can find the supported operators [here](https://getstream.io/chat/docs/ios-s
 
 #### Pagination
 
-You can fetch the next calls from the specified query, by calling the `loadNextCalls` method. For example, if you are doing pagination in SwiftUI, you can use the `onAppear` modifier on each entry, and based on its index fetch the next calls.
+You can encapsulae the querying calls login into an object `CallsViewModel` for simplicity and state management.
+```swift
+@MainActor
+final class CallsViewModel: ObservableObject {
+
+    @Injected(\.streamVideo) internal var streamVideo
+
+    @Published internal var calls = [Call]()
+
+    private var cancellables = Set<AnyCancellable>()
+
+    private lazy var callsController: CallsController = {
+        let sortParam = CallSortParam(direction: .descending, field: .createdAt)
+        let filters: [String: RawJSON] = ["type": .dictionary(["$eq": .string("audio_room")])]
+        let callsQuery = CallsQuery(sortParams: [sortParam], filters: filters, watch: true)
+        return streamVideo.makeCallsController(callsQuery: callsQuery)
+    }()
+
+    init() {
+        subscribeToCallsUpdates()
+        loadCalls()
+    }
+
+    func onCallAppear(_ call: Call) {
+        let index = calls.firstIndex { callData in
+            callData.cId == call.cId
+        }
+        guard let index else { return }
+
+        if index < calls.count - 10 {
+            return
+        }
+
+        loadCalls()
+    }
+
+    func loadCalls() {
+        Task {
+            try await callsController.loadNextCalls()
+        }
+    }
+
+    func subscribeToCallsUpdates() {
+        callsController.$calls.sink { calls in
+            DispatchQueue.main.async {
+                self.calls = calls
+            }
+        }
+        .store(in: &cancellables)
+    }
+}
+```
+
+You can then fetch the next calls from the specified query, by calling the `loadNextCalls` method. For example, if you are doing pagination in SwiftUI, you can use the `onAppear` modifier on each entry, and based on its index fetch the next calls.
 
 First, in your SwiftUI view, you can call a method from your presentation layer (for example a view model), on the view item appearance:
 
 ```swift
+let callsViewModel = CallsViewModel()
```

Comment:
shouldn't this be a @StateObject?

Reply:
Actually that shouldn't be there as this is a partial snippet. Integrators should have defined somewhere else the viewModel